zio-http/src/main/scala/zio/http/Body.scala

```diff
@@ -43,58 +45,83 @@ trait Body { self =>
 
   def isComplete: Boolean
 
-  def write(ctx: Ctx)(implicit trace: Trace): Task[Boolean]
 }
 
 object Body {
 
+  private[zio] sealed trait UnsafeWriteable extends Body
+
+  private[zio] sealed trait UnsafeBytes extends Body {
+    private[zio] def unsafeAsArray(implicit unsafe: Unsafe): Array[Byte]
+  }
+
   /**
    * Helper to create empty Body
    */
-  val empty: Body = new Body {
-    override def asChunk(implicit trace: Trace): Task[Chunk[Byte]]              = ZIO.succeed(Chunk.empty[Byte])
+  val empty: Body = EmptyBody
+
+  private[zio] object EmptyBody extends Body with UnsafeWriteable with UnsafeBytes {
+
+    override def asArray(implicit trace: Trace): Task[Array[Byte]] = ZIO.succeed(Array.empty)
```

Comment:
`= ZIO.succeedNow(Array.empty)`

Moreover, since empty arrays are effectively immutable, you can store the whole thing into a private `val` in the companion object (`val zioEmptyArray = ZIO.succeedNow(Array.empty[Byte])`).